### PR TITLE
Python version update

### PR DESCRIPTION
**Pull request by schandrika**

### CODE REVIEW

**Review by shwethanidd — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ VOLTTRON base historian framework that provide common functions such as caching,
 
 ## Requirements
 
- - Python >= 3.8
+ - Python >= 3.10
```
Comment (craigpnnl) [inline README.md:11]: Add poetry here...I need to do that on the core as well.

File: pyproject.toml [hunk 2]
```diff
@@ -11,8 +11,8 @@ keywords = []
 packages = [ { include = "historian", from = "src" }, { include = "historian", from = "tests" } ]
 
 [tool.poetry.dependencies]
-python = ">=3.8,<4.0"
-volttron = ">=10.0.2rc0,<11.0"
+python = ">=3.10,<4.0"
+volttron = ">=10.0.3a9,<11.0"
```
Comment (craigpnnl) [inline pyproject.toml:15]: This will be 10.0.4rc0 soon as the thing is released.

**Review by craigpnnl — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 2 above)
Comment (schandrika) [reply pyproject.toml:15]: won't poetry automatically pull the latest version that is within the given version range?

File: README.md (hunk 1 above)
Comment (schandrika) [reply README.md:11]: if core has poetry as requirement, that should do it right?

Comment (craigpnnl) [reply README.md:11]: Well then we should have VOLTTRON as a requirement right?  Poetry is really a development requirement...I suppose

File: pyproject.toml (hunk 2 above)
Comment (craigpnnl) [reply pyproject.toml:15]: yes...wondering if we want to be explicit though....I did notice that with the modbustk lib I have to update the base driver to volttron-testing 0.4 in order for it to be updated.

**Pull request merged**